core/mutex.c

```diff
@@ -58,6 +64,19 @@ int _mutex_lock(mutex_t *mutex, int blocking)
         else {
             thread_add_to_list(&mutex->queue, me);
         }
+
+#ifdef MODULE_CORE_PRIORITY_INHERITANCE
+        thread_t *owner = (thread_t *)thread_get(mutex->owner);
+        /* if my priority is higher than the priority of the current mutex
+         * owner, I lend my priority to the current owner */
+        if (me->priority < owner->priority) {
+            DEBUG("PID[%" PRIkernel_pid "]: changing priority from %i -> %i\n",
+                  owner->pid, owner->priority, me->priority);
+            mutex->prio_before = owner->priority;
+            sched_change_priority(owner, me->priority);
+        }
+#endif
+
         irq_restore(irqstate);
         thread_yield_higher();
         /* We were woken up by scheduler. Waker removed us from queue.
```

Comment:
Here you should then add the priority of the thread which currently has the mutex to `prio_before`. But doing it here might cause problems since IRQ are enabled, maybe is a better idea to do it in line 111 (see my comment below).